<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>SoundSelector</class>
 <widget class="QDialog" name="SoundSelector">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>524</width>
    <height>453</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Sound selector</string>
  </property>
  <layout class="QGridLayout" name="gridLayout">
   <item row="1" column="2">
    <widget class="QTreeWidget" name="rolesTree">
     <property name="statusTip">
      <string>Role tree</string>
     </property>
     <property name="whatsThis">
      <string>This is tree of associated and not associated roles (aliases) for sounds. Note that each role must be linked with one sound before table export. However you can use the workspace even although not all roles are reserved yet</string>
     </property>
     <property name="accessibleName">
      <string/>
     </property>
     <property name="styleSheet">
      <string notr="true">QTreeWidget {
	background-color: rgba(80, 200, 0, 0.1);	
}
</string>
     </property>
     <property name="frameShape">
      <enum>QFrame::StyledPanel</enum>
     </property>
     <property name="frameShadow">
      <enum>QFrame::Plain</enum>
     </property>
     <property name="alternatingRowColors">
      <bool>false</bool>
     </property>
     <property name="selectionMode">
      <enum>QAbstractItemView::ExtendedSelection</enum>
     </property>
     <property name="selectionBehavior">
      <enum>QAbstractItemView::SelectRows</enum>
     </property>
     <property name="uniformRowHeights">
      <bool>true</bool>
     </property>
     <property name="animated">
      <bool>false</bool>
     </property>
     <property name="columnCount">
      <number>1</number>
     </property>
     <attribute name="headerVisible">
      <bool>false</bool>
     </attribute>
     <attribute name="headerStretchLastSection">
      <bool>true</bool>
     </attribute>
     <column>
      <property name="text">
       <string>roles</string>
      </property>
     </column>
     <item>
      <property name="text">
       <string>Linked</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>Unlinked</string>
      </property>
     </item>
    </widget>
   </item>
   <item row="3" column="2">
    <widget class="QDialogButtonBox" name="buttonBox">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="standardButtons">
      <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set>
     </property>
    </widget>
   </item>
   <item row="0" column="0" rowspan="4">
    <widget class="QWidget" name="buttonWrapper" native="true">
     <property name="minimumSize">
      <size>
       <width>200</width>
       <height>0</height>
      </size>
     </property>
     <property name="maximumSize">
      <size>
       <width>200</width>
       <height>16777215</height>
      </size>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout">
      <item>
       <widget class="QPushButton" name="selectDirectoryButton">
        <property name="minimumSize">
         <size>
          <width>0</width>
          <height>30</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>16777215</width>
          <height>30</height>
         </size>
        </property>
        <property name="statusTip">
         <string>Select directory with sounds to be loaded</string>
        </property>
        <property name="whatsThis">
         <string>Click this button to evoke dialog window for selecting directory with WAV sounds to be loaded</string>
        </property>
        <property name="text">
         <string>Select directory</string>
        </property>
        <property name="icon">
         <iconset resource="icons.qrc">
          <normaloff>:/dialog_selector_icons/dialog_icons/sound_folder_icon.png</normaloff>:/dialog_selector_icons/dialog_icons/sound_folder_icon.png</iconset>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="selectSoundButton">
        <property name="minimumSize">
         <size>
          <width>0</width>
          <height>30</height>
         </size>
        </property>
        <property name="statusTip">
         <string>Select single sound or few sounds to load</string>
        </property>
        <property name="whatsThis">
         <string>Click this button to evoke dialog window for direct selection of single sound, or multiple set of sounds</string>
        </property>
        <property name="text">
         <string>Select sound(s)</string>
        </property>
        <property name="icon">
         <iconset resource="icons.qrc">
          <normaloff>:/dialog_selector_icons/dialog_icons/sounds_icon.png</normaloff>:/dialog_selector_icons/dialog_icons/sounds_icon.png</iconset>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="removeSelectedButton">
        <property name="minimumSize">
         <size>
          <width>0</width>
          <height>30</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>16777215</width>
          <height>30</height>
         </size>
        </property>
        <property name="statusTip">
         <string>Remove selected items in sound list</string>
        </property>
        <property name="whatsThis">
         <string>Click this button if you wish to delete selected items in imported sounds list</string>
        </property>
        <property name="text">
         <string>Remove selected</string>
        </property>
        <property name="icon">
         <iconset resource="icons.qrc">
          <normaloff>:/dialog_selector_icons/dialog_icons/sounds_remove_icon.png</normaloff>:/dialog_selector_icons/dialog_icons/sounds_remove_icon.png</iconset>
        </property>
       </widget>
      </item>
      <item>
       <widget class="Line" name="loadButtonsSeparator">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="linkButton">
        <property name="minimumSize">
         <size>
          <width>0</width>
          <height>35</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>16777215</width>
          <height>35</height>
         </size>
        </property>
        <property name="statusTip">
         <string>Links the selected sound with selected role</string>
        </property>
        <property name="whatsThis">
         <string>If single sound is selected in sound list and appropriate role is highlighted, sound will be associated with selected role and become hidden in sounds list</string>
        </property>
        <property name="text">
         <string>Link</string>
        </property>
        <property name="icon">
         <iconset resource="icons.qrc">
          <normaloff>:/dialog_selector_icons/dialog_icons/link_icon.png</normaloff>:/dialog_selector_icons/dialog_icons/link_icon.png</iconset>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="unlinkButton">
        <property name="minimumSize">
         <size>
          <width>0</width>
          <height>35</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>16777215</width>
          <height>35</height>
         </size>
        </property>
        <property name="statusTip">
         <string>Unlinks selected role with currently associated sound</string>
        </property>
        <property name="whatsThis">
         <string>If a role with associated sound is selected, click this button to break up the connection between them. Unlinked sound will appear back in sound list</string>
        </property>
        <property name="text">
         <string>Unlink</string>
        </property>
        <property name="icon">
         <iconset resource="icons.qrc">
          <normaloff>:/dialog_selector_icons/dialog_icons/unlink_icon.png</normaloff>:/dialog_selector_icons/dialog_icons/unlink_icon.png</iconset>
        </property>
       </widget>
      </item>
      <item>
       <widget class="Line" name="linkButtonsSeparator">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QWidget" name="playerWidget" native="true">
        <layout class="QGridLayout" name="gridLayout_2">
         <property name="leftMargin">
          <number>0</number>
         </property>
         <property name="topMargin">
          <number>0</number>
         </property>
         <property name="rightMargin">
          <number>0</number>
         </property>
         <property name="bottomMargin">
          <number>0</number>
         </property>
         <item row="0" column="0" colspan="2">
          <widget class="QLabel" name="fileToPlayLabel">
           <property name="styleSheet">
            <string notr="true">color: red; font-weight: bold; font-size: 16px; text-decoration: underline;</string>
           </property>
           <property name="text">
            <string>no file to play</string>
           </property>
           <property name="alignment">
            <set>Qt::AlignCenter</set>
           </property>
          </widget>
         </item>
         <item row="4" column="1">
          <widget class="QSlider" name="volumeSlider">
           <property name="minimumSize">
            <size>
             <width>0</width>
             <height>15</height>
            </size>
           </property>
           <property name="maximumSize">
            <size>
             <width>16777215</width>
             <height>15</height>
            </size>
           </property>
           <property name="statusTip">
            <string>Volume setting</string>
           </property>
           <property name="whatsThis">
            <string>This slider allows you to set up the volume level for current sound.  Volume level for other sounds will not be affected. If sound in imported sounds list is selected, this volume level represents the volume of all imported sounds and has no effect on associated sounds.</string>
           </property>
           <property name="maximum">
            <number>10</number>
           </property>
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
          </widget>
         </item>
         <item row="6" column="0">
          <widget class="QLabel" name="progressLabel">
           <property name="text">
            <string>Progress :</string>
           </property>
           <property name="alignment">
            <set>Qt::AlignCenter</set>
           </property>
          </widget>
         </item>
         <item row="8" column="0" colspan="2">
          <spacer name="verticalSpacer">
           <property name="orientation">
            <enum>Qt::Vertical</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>20</width>
             <height>40</height>
            </size>
           </property>
          </spacer>
         </item>
         <item row="1" column="0" colspan="2">
          <widget class="QLabel" name="durationLabel">
           <property name="styleSheet">
            <string notr="true">color: red;</string>
           </property>
           <property name="text">
            <string>Duration: 0 sec</string>
           </property>
           <property name="alignment">
            <set>Qt::AlignCenter</set>
           </property>
          </widget>
         </item>
         <item row="4" column="0">
          <widget class="QLabel" name="volumeLabel">
           <property name="text">
            <string>Volume (0):</string>
           </property>
           <property name="alignment">
            <set>Qt::AlignCenter</set>
           </property>
          </widget>
         </item>
         <item row="6" column="1">
          <widget class="QSlider" name="progressSlider">
           <property name="minimumSize">
            <size>
             <width>0</width>
             <height>15</height>
            </size>
           </property>
           <property name="maximumSize">
            <size>
             <width>16777215</width>
             <height>15</height>
            </size>
           </property>
           <property name="statusTip">
            <string>Progress bar</string>
           </property>
           <property name="whatsThis">
            <string>This slider represents the progress bar of sound currently being played.  You can use this slider to skip in sound file as well.</string>
           </property>
           <property name="styleSheet">
            <string notr="true"/>
           </property>
           <property name="maximum">
            <number>100</number>
           </property>
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="invertedControls">
            <bool>false</bool>
           </property>
           <property name="tickPosition">
            <enum>QSlider::NoTicks</enum>
           </property>
           <property name="tickInterval">
            <number>20</number>
           </property>
          </widget>
         </item>
         <item row="5" column="0" colspan="2">
          <widget class="QWidget" name="playerButtonWrapper" native="true">
           <property name="minimumSize">
            <size>
             <width>0</width>
             <height>40</height>
            </size>
           </property>
           <property name="maximumSize">
            <size>
             <width>16777215</width>
             <height>40</height>
            </size>
           </property>
           <layout class="QHBoxLayout" name="horizontalLayout">
            <item>
             <widget class="QPushButton" name="stopSoundButton">
              <property name="statusTip">
               <string>Stop current playing</string>
              </property>
              <property name="whatsThis">
               <string>Use this button to stop the currently played sound. Note that progress pointer will be returned to 0s state.</string>
              </property>
              <property name="text">
               <string>Stop</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="playSoundButton">
              <property name="statusTip">
               <string>Play current sound</string>
              </property>
              <property name="whatsThis">
               <string>Click this to play the sound currently selected.</string>
              </property>
              <property name="text">
               <string>Play</string>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="Line" name="previewSeparator">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QCheckBox" name="incrementalSwitchCheckbox">
        <property name="statusTip">
         <string>Incremental switching option on/off</string>
        </property>
        <property name="whatsThis">
         <string>This option allows to increment role item pointer to the next item if exists after association current role with sound</string>
        </property>
        <property name="layoutDirection">
         <enum>Qt::LeftToRight</enum>
        </property>
        <property name="text">
         <string>Hold last item index</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item row="2" column="2">
    <widget class="QListWidget" name="importedSoundsList">
     <property name="statusTip">
      <string>Image list</string>
     </property>
     <property name="whatsThis">
      <string>This list contains all sounds registered by program internally</string>
     </property>
     <property name="styleSheet">
      <string notr="true">QListWidget {
	background-color: rgba(80, 200, 0, 0.1);	
}</string>
     </property>
     <property name="selectionMode">
      <enum>QAbstractItemView::ExtendedSelection</enum>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <tabstops>
  <tabstop>selectDirectoryButton</tabstop>
  <tabstop>selectSoundButton</tabstop>
  <tabstop>removeSelectedButton</tabstop>
  <tabstop>linkButton</tabstop>
  <tabstop>unlinkButton</tabstop>
  <tabstop>volumeSlider</tabstop>
  <tabstop>stopSoundButton</tabstop>
  <tabstop>playSoundButton</tabstop>
  <tabstop>progressSlider</tabstop>
  <tabstop>incrementalSwitchCheckbox</tabstop>
  <tabstop>rolesTree</tabstop>
  <tabstop>importedSoundsList</tabstop>
 </tabstops>
 <resources>
  <include location="icons.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>SoundSelector</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>248</x>
     <y>254</y>
    </hint>
    <hint type="destinationlabel">
     <x>157</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>SoundSelector</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>316</x>
     <y>260</y>
    </hint>
    <hint type="destinationlabel">
     <x>286</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
